sgt-back-end/index.js

```diff
@@ -0,0 +1,137 @@
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(error => {
+      console.error(error)
+      res.status(500).json({ Error: 'An unexpected error has occurred.' })
+    });
+});
+
+app.post('/api/grades/', (req, res) => {
+  if (!req.body.name || !req.body.course || !req.body.score || req.body.score < 1 || req.body.score > 100) {
+    return res.status(400).json({ error: 'name, course, and score are required fields. Please fill in ALL of them. Scores must be 1-100.' });
+  }
+  const newContent = {
+    name: req.body.name,
+    course: req.body.course,
+    score: Number(req.body.score)
+  }
+
+  const sql = `
+      insert into "grades" ( "name", "course", "score")
+      values ($1, $2, $3)
+      returning *
+  `;
+  const arr = [newContent.name, newContent.course, newContent.score]
+  db.query(sql, arr)
+    .then(result => {
+      return res.status(201).json(result.rows[0])
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error has occurred!!' });
+    });
+});
+
+
+app.put('/api/grades/:gradeId', (req, res) => {  //updates current id if it exists
+  const id = Number(req.params.gradeId)
+  const updatedName = req.body.name;
+  const updatedCourse = req.body.course;
+  const updatedScore = Number(req.body.score);
+
+  if (!Number.isInteger(id) || id < 1 || isNaN(id)) {
+    return res.status(400).json({ error: `Invalid input, gradeId must be a positive integer only.` })
+  }
+  else if (!updatedCourse || !updatedName || !updatedScore) {
+    return res.status(400).json({ error: `The fields 'course', 'name', and 'score' are required` })
+  }
+  else if (updatedScore < 1 || updatedScore > 100) {
+    return res.status(400).json({ error: `scores must be between 1-100` })
+  }
+  const query = {
+    text: `
+      update "grades"
+        set "name" = $1,
+            "course" = $2,
+            "score" = $3
+      where "gradeId" = $4
+      returning *
+  `,
+    values: [updatedName, updatedCourse, updatedScore, id]
+  }
+  db.query(query)
+    .then(result => {
+      if (!result.rows[0]) {
+        res.status(404).json({ error: `Cannot find student with gradeId ${id} to update...` })
+      }
+      else if (result.rows[0]) {
+        res.status(200).json(result.rows[0])
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error has occurred!' });
+    });
+})
+
+
+
+app.delete('/api/grades/:gradeId', (req, res) => {
+
+  const id = Number(req.params.gradeId)
+  const updatedName = req.body.name;
+  const updatedCourse = req.body.course;
+  const updatedScore = Number(req.body.score);
+  if (!Number.isInteger(id) || id < 1 || isNaN(id)) {
+    return res.status(400).json({ error: `GradeId must be a positive integer only...` });
+  }
+  const query = {
+    text: `
+    delete from "grades"
+          where "gradeId" = ${id}
+    returning *;
+  `,
+    value: [id]
```

Comment:
Does `value` also work here or is it supposed to be `values`?

Reply:
Honest question, i honestly don't know 🤷

Reply:
it doesn't work if the property name is "values"  when the array has length of 1. I'm not sure why either. I just tried this a bunch of times 
![image](https://user-images.githubusercontent.com/64397693/101268385-073ac680-3717-11eb-86e0-30c02b2339b1.png)

Reply:
Interesting!